src/main/java/com/appraisal/entities/BaseEntity.java

```diff
@@ -0,0 +1,37 @@
+package com.appraisal.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.sql.Timestamp;
+import java.util.Calendar;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@MappedSuperclass
+public class BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private boolean deleted = false;
+
+    private Timestamp createdOn;
+
```

Comment:
You can use the annotation @CreationTimestamp and @UpdateTimestamp
instead of using prePersist or preUpdate